Review Wazuh logtest section (#6072)
wazuh/wazuh-documentation: source/user-manual/ruleset/testing.rst

         timestamp: 'Oct 15 21:07:00'
         hostname: 'linux-agent'
         program_name: 'sshd'

      **Phase 2: Completed decoding.
         name: 'sshd'
         parent: 'sshd'
         srcip: '18.18.18.18'
         srcport: '48928'
         srcuser: 'blimey'

      **Phase 3: Completed filtering (rules).
         id: '5710'
         level: '5'
         description: 'sshd: Attempt to login using a non-existent user'
         groups: '["syslog","sshd","authentication_failed","invalid_login"]'
         firedtimes: '1'
         gdpr: '["IV_35.7.d","IV_32.2"]'
         gpg13: '["7.1"]'
         hipaa: '["164.312.b"]'
         mail: 'false'
         mitre.id: '["T1110.001","T1021.004","T1078"]'
         mitre.tactic: '["Credential Access","Lateral Movement","Defense Evasion","Persistence","Privilege Escalation","Initial Access"]'
         mitre.technique: '["Password Guessing","SSH","Valid Accounts"]'
         nist_800_53: '["AU.14","AC.7","AU.6"]'
         pci_dss: '["10.2.4","10.2.5","10.6.1"]'
         tsc: '["CC6.1","CC6.8","CC7.2","CC7.3"]'
      **Alert to be generated.

The above result shows that rule id ``5710`` matches the event log.

> If you paste the log seven more times within two minutes, you can see that rule id ``5710`` matches multiple times. You can see that in *Phase 3, filtering (rules)*, the ``firedtimes`` counter increases with each repetition. But for the last log line, rule id ``5712`` makes the match. This rule captures the eighth event that was matching rule id ``5710`` previously for the same IP address.

Reviewer: ```suggestion
If you paste the log seven more times within two minutes, you can see that rule id ``5710`` matches multiple times. You can see that in *Phase 3, filtering (rules)*, the ``firedtimes`` counter increases with each repetition. But for the last log line, rule id ``5712`` makes the match. This rule captures the eighth event that rule id ``5710`` matched previously for the same IP address.
```

> If you paste the log seven more times within two minutes, you can see that rule id ``5710`` matches multiple times. You can see that in *Phase 3, filtering (rules)*, the ``firedtimes`` counter increases with each repetition. But for the last log line, rule id ``5712`` makes the match. This rule captures the eighth event that was matching rule id ``5710`` previously for the same IP address.

Reviewer: ```suggestion
If you paste the log seven more times within two minutes, you can see that rule id ``5710`` matches multiple times and that in *Phase 3, filtering (rules)*, the ``firedtimes`` counter increases with each repetition. But for the last log line, rule id ``5712`` makes the match. This rule captures the eighth event that was matching rule id ``5710`` previously for the same IP address.
```

Author: We need shorter sentences

-  **log_format**: ``syslog`` or ``json``
-  **location**: path string
-  **event**: string

Logging into the Wazuh API
^^^^^^^^^^^^^^^^^^^^^^^^^^

Wazuh API endpoints require authentication. All calls must include a JSON Web Token. You can use the cURL command to log in. The Wazuh API provides a JWT token upon success.

#. Run the following command replacing ``<WAZUH_API_USER>`` and ``<PASSWORD>`` with your own values.

   .. code-block:: bash

      TOKEN=$(curl -u <WAZUH_API_USER>:<PASSWORD> -k -X POST "https://localhost:55000/security/user/authenticate?raw=true")

#. Check that everything works correctly.

   .. code-block:: bash

      curl -k -X GET "https://localhost:55000/?pretty=true" -H "Authorization: Bearer $TOKEN"

   .. code-block:: JSON
      :class: output

      {
         "data": {
            "title": "Wazuh API REST",
            "api_version": "4.4.1",
            "revision": 40406,
            "license_name": "GPL 2.0",
            "license_url": "https://github.com/wazuh/wazuh/blob/v4.4.1/LICENSE",
            "hostname": "centos7",
            "timestamp": "2023-04-25T13:39:23Z"
         },
         "error": 0
      }

First request
^^^^^^^^^^^^^

> The first time you send a processing request it has no logtest session token. Since there is no active session, a processing

Reviewer: ```suggestion
The first time you send a processing request, it has no logtest session token. Since there is no active session, a processing
```

log request is sent to logtest in Analysisd.

> Use the following sample data for request.

Reviewer: ```suggestion
Use the following sample data for request:
```

+------------------+--------------------------------------+----------------------------------------------------------------------------------------------+
| Field            | Description                          | Example                                                                                      |
+==================+======================================+==============================================================================================+

> | log_format       | Type of log, ``syslog`` or ``json``  | ``syslog``                                                                                   |

Reviewer 2: The [Logtest documentation](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.logtest_controller.run_logtest_tool) indicates that the endpoint accepts the following log formats: syslog, json, snort-full, squid, eventlog, eventchannel, audit, mysql_log, postgresql_log, nmapg, iis, command, full_command, djb-multilog, multi-line.

+------------------+--------------------------------------+----------------------------------------------------------------------------------------------+
| event            | Log to process                       | ``Oct 15 21:07:00 linux-agent sshd[29205]: Invalid user blimey from 18.18.18.18 port 48928`` |
+------------------+--------------------------------------+----------------------------------------------------------------------------------------------+
| location         | Origin of the log                    | ``master->/var/log/syslog``                                                                  |
+------------------+--------------------------------------+----------------------------------------------------------------------------------------------+
| token (optional) | logtest session id                   |                                                                                              |
+------------------+--------------------------------------+----------------------------------------------------------------------------------------------+

> You must send the data to the logtest endpoint in JSON format. You can first store the request in a variable as follows.

Reviewer: ```suggestion
You must send the data to the logtest endpoint in JSON format. You can first store the request in a variable as follows:
```

               "full_log": "Oct 15 21:07:00 linux-agent sshd[29205]: Invalid user blimey from 18.18.18.18 port 48928",
               "predecoder": {
                  "program_name": "sshd",
                  "timestamp": "Oct 15 21:07:00",
                  "hostname": "linux-agent"
               },
               "decoder": {
                  "parent": "sshd",
                  "name": "sshd"
               },
               "data": {
                  "srcip": "18.18.18.18",
                  "srcport": "48928",
                  "srcuser": "blimey"
               },
               "location": "master->/var/log/syslog"
            },
            "alert": true,
            "codemsg": 0
         }
      }

The above result shows that rule id ``5710`` matches the event log.

The ``messages`` field shows the session token ``95375d4c``. You must add this token to the next requests to keep the session loaded, including its event history, and rules and decoders. If you don't add the token field to the next request, a new session initializes, reloading the rules and decoders.

Repeating the request with the same session
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Add the session token to the request and send it seven more times within two minutes. You can see that rule id ``5710`` matches multiple times. In the ``rule`` object of the response, inside the ``output`` field, you can see the ``firedtimes`` counter increases with each repetition. But for the last request, rule id ``5712`` makes the match. This rule captures the eighth event that was matching rule id ``5710`` previously for the same IP address.

Reviewer: ```suggestion
Add the session token to the request and send it seven more times within two minutes. You can see that rule id ``5710`` matches multiple times. In the ``rule`` object of the response, inside the ``output`` field, you can see the ``firedtimes`` counter increases with each repetition. But for the last request, rule id ``5712`` makes the match. This rule captures the eighth event that rule id ``5710`` matched previously for the same IP address.
```
